src/io.c

```diff
@@ -5,6 +5,39 @@
 
 io_state_t io_state;
 
+#if defined(TARGET_NANOS)
+static timeout_callback_fn_t* timeout_cb;
+
+void nanos_clear_timer()
+{
+	timeout_cb = NULL;
+}
+
+void nanos_set_timer(int ms, timeout_callback_fn_t* cb)
+{
+	// if TRACE() is enabled, set_timer must be called
+	// before ui_ methods, because it causes Ledger Nano S
+	// to freeze in debug mode
+	// TRACE();
+	ASSERT(timeout_cb == NULL);
+	ASSERT(ms >= 0);
+	timeout_cb = cb;
+	UX_CALLBACK_SET_INTERVAL((unsigned) ms);
+}
+
+#define HANDLE_UX_TICKER_EVENT(ux_allowed) \
+	{\
```

Comment:
Small detail: Can we turn this to `do {} while(0)` macro style? http://bruceblinn.com/linuxinfo/DoWhile.html